[CCIP-2942] OCR2 plugins merge fixes [CCIP-2942]

asoliman92 opened the pull request:
Fix ccip ocr2 plugins fixes. Including missing files, tests and editing shared files. This needs a good review before merging. Should be merged into [ccip/ocr2-merge-clean](https://github.com/smartcontractkit/chainlink/tree/ccip/ocr2-merge-clean) with Squashing commits.

asoliman92 committed:
Add status checker

original commit:
https://github.com/smartcontractkit/ccip/commit/451984ad0469c7d685e305dba0a0d94eb0c9a053

asoliman92 committed:
Add CCIP to types.go

asoliman92 committed:
Add UnsafeSetConnectionsManager to feeds service for testing

asoliman92 committed:
Add CCIP feature to core.toml

mateusz-sekara committed:
Rebuilding config

mateusz-sekara committed:
Wiring up relayer and ocr2 delegates - this commit touches shared code !

mateusz-sekara committed:
Adding mockery configuration for ccip specific code

mateusz-sekara committed:
Setting CCIP feature flag to true - it's no longer used anywhere

makramkd commented on core/scripts/go.mod:
```diff
@@ -60,7 +60,7 @@ require (
 	github.com/XSAM/otelsql v0.27.0 // indirect
 	github.com/armon/go-metrics v0.4.1 // indirect
-	github.com/avast/retry-go/v4 v4.5.1 // indirect
+	github.com/avast/retry-go/v4 v4.6.0 // indirect
```
Whats the reasoning behind this bump

asoliman92 replied:
It's bumped in CCIP, Otherwise it will fail the build

makramkd commented on core/services/feeds/service.go:
```diff
@@ -107,6 +107,9 @@ type Service interface {
 	RejectSpec(ctx context.Context, id int64) error
 	UpdateSpecDefinition(ctx context.Context, id int64, spec string) error
+
+	// UnsafeSetConnectionsManager Only for testing
+	UnsafeSetConnectionsManager(ConnectionsManager)
```
Isn't this Unsafe_SetConnectionsManager? (note the underscore)

asoliman92 replied:
Chainlink code didn't have it. I re-added it manually. Didn't use the `_` because was getting some linting messages locally AFAIR. Does the underscore mean something in this context?

makramkd replied:
It doesn't change the meaning of the func but just wondering if this'll botch up any future merges from ccip -> chainlink

makramkd commented on core/services/ocr2/plugins/ccip/testhelpers/integration/chainlink.go:
```diff
@@ -378,7 +379,6 @@ func setupNodeCCIP(
 			fmt.Sprintf("127.0.0.1:%d", port),
 		}
 		c.Log.Level = &loglevel
-		c.Feature.CCIP = &trueRef
```
Intended?

asoliman92 replied:
hmmmm... I think this happened during merging.  @mateusz-sekara you changed something with the flag. We still expect it to be set though. right?

mateusz-sekara replied:
This flag is not used anywhere, but we need to deprecate it first and then remove it in the following release. To align it with how it's done in other places, changing the default value to true. This is what we've agreed on with @jmank88

mateusz-sekara replied:
(we can't remove it immediately because the parser is strict and it needs to be removed first from all the deployments)

asoliman92 committed:
Rename function

 rename UnsafeSetConnectionsManager to Unsafe_SetConnectionsManager to follow same convention in ccip repo

winder reviewed (approved):
Changes seem fine to me. I didn't diff every file against the ccip repo, but the ones I did look pretty much identical.
